stdpopsim/dfe.py

```diff
@@ -5,40 +5,187 @@
 import textwrap
 import attr
 import collections.abc
-from . import ext
+import numpy as np
+
+
+@attr.s(kw_only=True)
+class MutationType(object):
+    """
+    Class representing a "type" of mutation.  The design closely (exactly)
+    mirrors SLiM's MutationType class.
+
+    The main thing that mutation types carry is a way of drawing a selection
+    coefficient for each new mutation. This ``distribution_type`` should be one
+    of (see the SLiM manual for more information on these):
+
+    - ``f``: fixed, one parameter (the selection coefficient)
+    - ``e``: exponential, one parameter (mean)
+    - ``g``: gamma, two parameters (mean, shape)
+    - ``n``: normal, two parameters (mean, SD)
+    - ``w``: Weibull, two parameters (scale, shape)
+    - ``l``: logNormal, two parameters (mean and sd on log scale; see rlnorm)
+
+    :ivar distribution_type: A one-letter abbreviation for the distribution of
+        fitness effects that each new mutation of this type draws from (see below).
+    :vartype distribution_type: str
+    :ivar distribution_args: Arguments for the distribution type.
+    :vartype distribution_type: list
+    :ivar dominance_coeff: A single dominance coefficient (negative = underdominance,
+        0 = recessive, 0.5 = additive, 1.0 = completely dominant, > 1.0 = overdominant)
+    :vartype dominance_coeff: float
+    :ivar convert_to_substitution: Whether to retain any fixed mutations in the
+        simulation: otherwise, they have no effect on fitness once fixed (but
```

Comment:
Don't they still have no effect on fitness once fixed, because fitness is relative to other individuals?

Reply:
Right, that is true. How about
```
    :ivar convert_to_substitution: Whether to retain any fixed mutations in the
        simulation: if not, we cannot ask about their frequency once fixed.
        (Either way, they will remain in the tree sequence).

```